_docs-sources/pipelines/upgrading/index.md

```diff
@@ -0,0 +1,166 @@
+
+## Prerequisites
+
+- [Boilerplate](https://github.com/gruntwork-io/boilerplate#install) installed on your system (requires Gruntwork subscription)
+- Ability to create repositories in your GitHub Organization
+- Ability to add users to your GitHub Organization and the Gruntwork Developer Portal
+- Permissions to create secrets in GitHub repositories
+
+## Token Setup
+
+Modern Gruntwork Pipelines uses several tokens across workflows, with each token possessing only the minimal set of permissions necessary for its specific task. To uphold the principle of least privilege for each token, we advise maintaining two distinct users: the pre-existing CI user and a new CI Read Only user. The CI Read Only user should have access to read your infrastructure-live repository, infrastructure-modules repository (if applicable), and the Gruntwork Library.
+
+### Creating a CI Read Only User
+
+Follow the steps on [signing up a new GitHub account](https://docs.github.com/en/get-started/signing-up-for-github/signing-up-for-a-new-github-account) to create a new user. If you have an e-mail address for your existing CI user, we recommend using plus addressing, such as `ci+read-only@your-domain.com`, as the e-mail address for the new account.
+
+Add the new user to your GitHub org, adding it to a Team with read only access to your `infrastructure-live` and `infrastructure-modules` repository (if applicable). Follow the steps on [adding a user to the Gruntwork Developer portal](../../developer-portal/invite-team.md) to grant the CI Read Only user access to the Gruntwork Library.
+
+### Creating tokens for Workflows
+
+- `GRUNTWORK_CODE_ACCESS_TOKEN`: Classic token with READ access to repos. This should should be associated with the `CI Read Only` user created in [creating a CI Read only user](#creating-a-ci-read-only-user).
+- `PIPELINES_DISPATCH_TOKEN`: Fine Grained token that only has workflow permissions to execute workflows in the `infrastructure-pipelines` repository. This token should be associated with your existing CI user.
+- `INFRA_LIVE_ACCESS_TOKEN`: Fine grained token with READ and WRITE access to your `infrastructure-live` repository. This token should be associated with your existing CI user.
+
+## Creating your infrastructure-pipelines repository
+
+Modern Gruntwork Pipelines employ a dual-repository approach to isolate infrastructure as code (IaC) code definitions and IaC deployment mechanisms. This approach signifies that a single, highly secured repository, with write access limited to a specific group of administrators, contains both the code responsible for deploying your IaC and the access to your AWS accounts. Workflows within this repository leverage OpenId Connect (OIDC) and AWS IAM roles for creating short-lived session credentials, as opposed to storing long-lived credentials as secrets.
```

Comment:
"employ" here implies that `Modern Gruntwork Pipelines` is plural, in other places in this document it's interpreted as a singular product name. See line 16 "Gruntwork Pipelines uses". We should be consistent. I'm more inclined to treat it as a singular product name, but happy to be overruled.

Reply:
Ah good catch, thank you!

Reply:
Fixed in [73295b2](https://github.com/gruntwork-io/docs/pull/1115/commits/73295b26e2fbe7ab38f0addf4381fb0d3a67be12)